contracts/adapters/FinancingChainlink.sol

```diff
@@ -0,0 +1,185 @@
+        require(
+            isNotReservedAddress(applicant),
+            "applicant using reserved address"
+        );
+
+        dao.submitProposal(proposalId);
+
+        ProposalDetails storage proposal = proposals[address(dao)][proposalId];
+        proposal.applicant = applicant;
+        proposal.amount = amount;
+        proposal.token = token;
+
+        IVoting votingContract = IVoting(dao.getAdapterAddress(VOTING));
+        address sponsoredBy =
+            votingContract.getSenderAddress(
+                dao,
+                address(this),
+                data,
+                msg.sender
+            );
+
+        dao.sponsorProposal(proposalId, sponsoredBy, address(votingContract));
+        votingContract.startNewVotingForProposal(dao, proposalId, data);
+    }
+
+    /**
+     * @notice Processing a financing proposal to grant the requested funds.
+     * @dev Only proposals that were not processed are accepted.
+     * @dev Only proposals that were sponsored are accepted.
+     * @dev Only proposals that passed can get processed and have the funds released.
+     * @param dao The DAO Address.
+     * @param proposalId The proposal id.
+     */
+    function processProposal(DaoRegistry dao, bytes32 proposalId)
+        external
+        override
+        reentrancyGuard(dao)
+    {
+        ProposalDetails memory details = proposals[address(dao)][proposalId];
+
+        IVoting votingContract = IVoting(dao.votingAdapter(proposalId));
+        require(address(votingContract) != address(0), "adapter not found");
+
+        require(
+            votingContract.voteResult(dao, proposalId) ==
+                IVoting.VotingState.PASS,
+            "proposal needs to pass"
+        );
+        dao.processProposal(proposalId);
+        BankExtension bank = BankExtension(dao.getExtensionAddress(BANK));
+
+        //enter US Dollar = details.amount to convert it into ETH
+        uint256 amount = _usdToEth(details.amount);
+
+        bank.subtractFromBalance(GUILD, details.token, amount);
+        bank.addToBalance(details.applicant, details.token, amount);
+    }
+
+    /**
+     *  Returns the latest price from AggregatorV3Interface.sol
+        Chainlink returns the latest price to 5 decimials 
+        e.g. 176471000000 = $1,764.71
+     */
+    function getLatestPrice() public view returns (int256) {
+        (
+            uint80 roundId,
+            int256 answer,
+            uint256 startedAt,
+            uint256 updatedAt,
+            uint80 answeredInRound
+        ) = _priceFeed.latestRoundData();
+        return answer;
+    }
+
+    /* 
+        @param _amount = the amount in US Dollars to convert to ETH
+        multipliers are needed in numerator (usd to wei) and denominator 
+        (convert result from chainlink to wei) 
+     */
+    function _usdToEth(uint256 _amount) internal view returns (uint256) {
+        //convert int to uint
+        uint256 denominator = uint256(getLatestPrice());
```

Comment:
was wondering if it makes sense to replace the `getLatestPrice()` call by the `_priceFeed.latestRoundData()`, and then get the `answer`, so we don't need to do an additional call to a public function to get the price. WDYT?

Reply:
move `priceFeed.latestRoundData()` and moving it inside of the _`usdToEth()` function might work.  I'm trying to think of a scenario where we may want to keep the two separate.